snark-verifier-sdk/Cargo.toml

```diff
@@ -0,0 +1,55 @@
+[package]
+name = "snark-verifier-sdk"
+version = "0.1.1"
+edition = "2021"
+
+[dependencies]
+itertools = "0.10.3"
+lazy_static = "1.4.0"
+num-bigint = "0.4.3"
+num-integer = "0.1.45"
+num-traits = "0.2.15"
+rand = "0.8"
+rand_chacha = "0.3.1"
+hex = "0.4"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+bincode = "1.3.3"
+ark-std = { version = "0.3.0", features = ["print-trace"], optional = true }
+
+snark-verifier = { path = "../snark-verifier", default-features = false }
+
+# system_halo2
+halo2_proofs = { git = "https://github.com/privacy-scaling-explorations/halo2", tag = "v2023_04_20" } # not optional for now
+halo2curves = { git = 'https://github.com/privacy-scaling-explorations/halo2curves', tag = "0.3.2" } # must be same version as in halo2_proofs
+
+# loader_halo2
+halo2_wrong_ecc = { git = "https://github.com/privacy-scaling-explorations/halo2wrong", tag = "v2023_04_20", package = "ecc", optional = true }
+poseidon = { git = "https://github.com/privacy-scaling-explorations/poseidon", tag = "v2023_04_20", optional = true }
+
+# loader_evm
+ethereum-types = { version = "0.14", default-features = false, features = ["std"], optional = true }
+
+[dev-dependencies]
+ark-std = { version = "0.3.0", features = ["print-trace"] }
+paste = "1.0.7"
+pprof = { version = "0.11", features = ["criterion", "flamegraph"] }
+criterion = "0.4"
+criterion-macro = "0.4"
+# loader_evm
+crossterm = { version = "0.25" }
+tui = { version = "0.19", default-features = false, features = ["crossterm"] }
+
+[features]
+default = ["loader_halo2", "loader_evm", "derive_serde", "display"]
+display = ["dep:ark-std"]
+loader_evm = ["snark-verifier/loader_evm", "dep:ethereum-types"]
+loader_halo2 = ["snark-verifier/system_halo2", "snark-verifier/loader_halo2", "dep:halo2_wrong_ecc", "dep:poseidon"]
+parallel = ["snark-verifier/parallel"]
+derive_serde = ["snark-verifier/derive_serde", "halo2curves/derive_serde"]
+halo2_circuit_params = ["snark-verifier/halo2_circuit_params"]
```

Comment:
I feel we could disable the `halo2_circuit_param` for now if it's not so useful yet, otherwise the `read_pk` and
`gen_pk` in `snark-verifier-sdk/src/lib.rs` need to be updated to pass the CI.

Reply:
I commented it out. Does the CI check --all-features or something?